components/promise-component.js

```diff
@@ -0,0 +1,66 @@
+// promise-component.js
+
+var _ = require('underscore');
+var noflo = require('noflo');
+
+module.exports = function(def){
+    if (_.isEmpty(def.inPorts)) throw Error("Missing inPorts");
+    if (def.resolvePort && !_.isString(def.resolvePort.name))
+        throw Error("Missing resolvePort name");
+    if (def.rejectPort && !_.isString(def.rejectPort.name))
+        throw Error("Missing rejectPort name");
+    var resolvePort = def.resolvePort || _.defaults({
```

Comment:
This seems odd - why do you check to see if the reject/resolve port is defined and has a name above (and throw error if not), and then set these defaults on it below with a hard coded default name?